### DISCUSSION TITLE
Is it possible to visualize tunnels?

### DISCUSSION THREAD

**Discussion by marus16** [molstar/molstar#265] [Q&A] [answered]
Hello, 

I have a question related to tunnels. We use PQR files with coordinates and radius to visualize tunnels. I've seen here in issues (https://github.com/molstar/molstar/issues/157#issue-848983361) that PQR files are not supported. I was able to visualize tunnels when loading PQR as PDB file but tunnel, as expected, is visualized as usual atoms. I would like to ask if there is any possibility to load tunnels (maybe in different format) and if there is any plan to implement the support of PQR files in the future? 
Thank you for your reply :)

**Comment by dsehnal** [comment, marked as answer]
I suppose we could support PRQ files in the future. As for displaying the tunnels, you would have to write your own visual that renders that atoms as spheres with the corresponding radius or constructs the appropriate iso-surface.

**Reply by marus16** [reply]
Thank you for your help and advice :)

**Reply by marus16** [reply]
Hello, 

I have a followup question. I was able to visualize tunnels as spheres but I need tunnel to be transparent in some cases. I managed to make spheres transparent but the issue is in intersecting parts. Is it somehow possible to visualize spheres as one transparent object and not to see intersecting parts? If yes, can you please advise me how to do it? And if not, is there another way how to possibly solve this? Maybe using the appropriate iso-surface which you suggested as a second solution? 

Thank you for your help :)

**Reply by dsehnal** [reply]
Hey, try search the source for ``computeMarchingCubesMesh`` and ``GaussianDensityCPU``. Basically you can create ``PositionData`` from your tunnel profile and pass that to ``GaussianDensityCPU`` to get the density grid which you can triangulate using ``computeMarchingCubesMesh``. What software are you using to compute the tunnels?

**Reply by marus16** [reply]
Thanks, I'll try it. For the computation of tunnels we use Caver.
